src/main/scala/com/microsoft/hyperspace/actions/RefreshIncrementalAction.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.actions
+
+import org.apache.spark.sql.{SaveMode, SparkSession}
+import org.apache.spark.sql.functions.col
+
+import com.microsoft.hyperspace.HyperspaceException
+import com.microsoft.hyperspace.index._
+import com.microsoft.hyperspace.index.DataFrameWriterExtensions.Bucketizer
+import com.microsoft.hyperspace.telemetry.{AppInfo, HyperspaceEvent, RefreshIncrementalActionEvent}
+
+/**
+ * Action to create indexes on newly arrived data. If the user appends new data to existing,
```

Comment:
To reviewers:
1) This file is copied from RefreshAppendAction.scala and I haven't revised overall comments yet. I'll update the comments later.
2) I haven't removed RefreshAppendAction.scala and RefreshDeleteAction.scala yet, for reference.
I'll remove them and events later.

Reply:
This approach looks good. Let's move forward with this!